lib/matplotlib/backends/backend_qt5.py

```diff
@@ -782,97 +782,47 @@ class SubplotToolQt(SubplotTool, UiSubplotTool):
-
-    def funcbottom(self, val):
-        if val == self.slidertop.value():
-            val -= 1
-        val /= 1000.
-        self.targetfig.subplots_adjust(bottom=val)
-        self.bottomvalue.setText("%.2f" % val)
-        if self.drawon:
-            self.targetfig.canvas.draw_idle()
-
-    def functop(self, val):
-        if val == self.sliderbottom.value():
-            val += 1
-        val /= 1000.
-        self.targetfig.subplots_adjust(top=val)
-        self.topvalue.setText("%.2f" % val)
+        self._figure = targetfig
+
+        for lower, higher in [("bottom", "top"), ("left", "right")]:
+            self._widgets[lower].valueChanged.connect(
+                lambda val: self._widgets[higher].setMinimum(val + .001))
+            self._widgets[higher].valueChanged.connect(
+                lambda val: self._widgets[lower].setMaximum(val - .001))
+
+        self.defaults = {
+            attr: getattr(self._figure.subplotpars, attr)
+            for attr in ["left", "bottom", "right", "top", "wspace", "hspace"]}
+        # Set values after setting the range callbacks, but before setting up
+        # the redraw callbacks.
+        self._reset()
+
+        for attr in self.defaults:
+            self._widgets[attr].valueChanged.connect(self._on_value_changed)
+        for action, method in [("Tight Layout", self._tight_layout),
+                               ("Reset", self._reset),
+                               ("Close", self.close)]:
+            self._widgets[action].clicked.connect(method)
+
+    def _on_value_changed(self):
+        self._figure.subplots_adjust(
+            **{attr: self._widgets[attr].value() for attr in self.defaults})
         if self.drawon:
-            self.targetfig.canvas.draw_idle()
-
-    def funcwspace(self, val):
-        val /= 1000.
-        self.targetfig.subplots_adjust(wspace=val)
-        self.wspacevalue.setText("%.2f" % val)
-        if self.drawon:
-            self.targetfig.canvas.draw_idle()
-
-    def funchspace(self, val):
-        val /= 1000.
-        self.targetfig.subplots_adjust(hspace=val)
-        self.hspacevalue.setText("%.2f" % val)
+            self._figure.canvas.draw_idle()
+
+    def _tight_layout(self):
+        self._figure.tight_layout()
+        for attr in self.defaults:
+            widget = self._widgets[attr]
+            widget.blockSignals(True)
+            widget.setValue(getattr(self._figure.subplotpars, attr))
+            widget.blockSignals(False)
         if self.drawon:
-            self.targetfig.canvas.draw_idle()
-
-    def functight(self):
-        self.targetfig.tight_layout()
-        self._setSliderPositions()
-        self.targetfig.canvas.draw_idle()
+            self._figure.canvas.draw_idle()
 
-    def reset(self):
-        self.targetfig.subplots_adjust(**self.defaults)
-        self._setSliderPositions()
-        self.targetfig.canvas.draw_idle()
+    def _reset(self):
+        for attr in self.defaults:
```

Comment:
`self.defaults` keeps being used simply as a source of the list of parameters; for the sake of readability, I might replace those uses with a genuine list, e.g. `self.params`.  Yes, it would add one more line of code to the module.